bundles/org.openhab.binding.phc/src/main/java/org/openhab/binding/phc/internal/handler/PHCBridgeHandler.java

```diff
@@ -384,12 +583,39 @@ private void sendJRM(byte moduleAddress, byte channel, Command command, short up
                 break;
         }
 
-        serialWrite(buildMessage(moduleAddress, channel, cmd, toggleChannel(moduleAddress, channel)));
+        serialWrite(buildMessage(module, channel, cmd, getToggle(module)));
     }
 
-    private void sendPorBroadcast(byte b) {
-        byte[] cmd = { 0 };
-        writeMsg(buildMessage((byte) 0xFF, 0, cmd, false), 20); // 20 times needed?
+    private void sendDim(byte moduleAddress, byte channel, Command command, short dimTime) {
+        byte module = (byte) (moduleAddress | 0xA0);
+        byte[] cmd = new byte[(command instanceof PercentType && !(((PercentType) command).byteValue() == 0)) ? 3 : 1];
+
+        cmd[0] = (byte) (channel << 5);
+
+        if (command instanceof OnOffType) {
+            if (command.equals(OnOffType.ON)) {
```

Comment:
Position literals first in `String` comparisons - that way if the `String` is `null` you won't get a `NullPointerException`, it'll just return `false`. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)

Reply:
Check all please